ucte/ucte-converter/src/main/java/com/powsybl/ucte/converter/UcteImporter.java

```diff
@@ -497,55 +522,95 @@ private static void createRatioTapChanger(UctePhaseRegulation uctePhaseRegulatio
             double dy = i * ucteAngleRegulation.getDu() / 100 * Math.sin(Math.toRadians(ucteAngleRegulation.getTheta()));
-            switch (ucteAngleRegulation.getType()) {
-                case ASYM:
-                    rho = 1d / Math.hypot(dy, 1d + dx);
-                    alpha = Math.toDegrees(Math.atan2(dy, 1 + dx));
-                    break;
-
-                case SYMM:
-                    rho = 1d;
-                    alpha = Math.toDegrees(2 * Math.atan2(dy, 2 * (1d + dx)));
-                    break;
-
-                default:
-                    throw new AssertionError("Unexpected UcteAngleRegulationType value: " + ucteAngleRegulation.getType());
-            }
+            Pair<Double, Double> rhoAndAlpha = getRhoAndAlpha(ucteAngleRegulation, dx, dy, currentRatioTapChangerRho);
             ptca.beginStep()
-                    .setRho(rho)
-                    .setAlpha(-alpha) // minus because in the UCT model PST is on side 2 and side1 on IIDM model
+                    .setRho(rhoAndAlpha.getLeft())
+                    .setAlpha(-rhoAndAlpha.getRight()) // minus because in the UCT model PST is on side 2 and side1 on IIDM model
                     .setR(0)
                     .setX(0)
                     .setG(0)
                     .setB(0)
                     .endStep();
         }
+        return ptca;
+    }
+
+    private static void createPhaseTapChanger(UcteAngleRegulation ucteAngleRegulation, TwoWindingsTransformer transformer) {
+        LOGGER.trace("Create phase tap changer '{}'", transformer.getId());
+        PhaseTapChangerAdder ptca = createPhaseTapChangerAdder(ucteAngleRegulation, transformer, null);
+        ptca.add();
+    }
+
+    private static Pair<Double, Double> getRhoAndAlpha(UcteAngleRegulation ucteAngleRegulation, double dx, double dy,
+                                                       Double currentRatioTapChangerRho) {
+        if (currentRatioTapChangerRho != null && currentRatioTapChangerRho.equals(0.0)) {
+            throw new AssertionError("Unexpected non zero value for current ratio tap changer rho: " + currentRatioTapChangerRho);
+        }
+        double rho;
+        double alpha;
+        switch (ucteAngleRegulation.getType()) {
+            case ASYM:
+                if (currentRatioTapChangerRho == null) {
+                    rho = 1d / Math.hypot(dy, 1d + dx);
+                    alpha = Math.toDegrees(Math.atan2(dy, 1 + dx));
+                } else {
+                    double dyEq = dy / currentRatioTapChangerRho;
+                    double dxEq = (dx + 1) / currentRatioTapChangerRho - 1;
+                    rho = 1d / Math.hypot(dyEq, 1d + dxEq) / currentRatioTapChangerRho; // the formula = 1d / Math.hypot(dyEq, 1d + dxEq) already takes into account rhoInit, so we divide by rhoInit that will be carried by the ratio tap changer
+                    alpha = Math.toDegrees(Math.atan2(dyEq, 1 + dxEq));
```

Comment:
I don't understand what is trying to be achieved here.

Using the equations written here:
rho = 1 / hypot(dyEq, 1+dxEq) / rtcRho
      = 1 / hypot(dy/rtcRho, (dx+1)/rtcRho) / rtcRho
      = 1 / hypot(dy, dx+1) / rtcRho²
and 
alpha = atan2(dyEq, 1+dxEq)
          = atan2(dy/rtcRho, (dx+1)/rtcRho)
          = atan2(dy, dx+1)

Are these really the equations we wanted to use? (alpha would be unchanged, and rho would be divided by rtcRho squared)

Reply:
you are right, something is wrong, I will check that

Reply:
by the way here, rho is not divided by rtcRho squared but is unchanged